File: implementations/php/src/RCL.php
<?php
final class RCL {
  private static function invalid(string $text): void {
    if (preg_match('/=\s*\'.*\'/m', $text)) throw new RuntimeException('single-quoted string');
    if (preg_match('/,\s*\]/m', $text)) throw new RuntimeException('trailing comma in array');
    if (preg_match('/=\s*([A-Za-z_][A-Za-z0-9_]*)\s*$/m', $text, $m) && !in_array($m[1], ['true','false'], true)) throw new RuntimeException('invalid bare value');
  }
  private static function project(string $text): array {
    self::invalid($text);
    $name = null; $region = null;
    if (preg_match('/region\s+"([^"]+)"\s+do[\s\S]*?name\s*=\s*"([^"]+)"/m', $text, $m)) { $region = $m[1]; $name = $m[2]; }
    if (preg_match('/config\s+do/m', $text) && $region !== null) return ['config' => ['regions' => [$region => ['name' => $name]]]];
    return [];
  }
  public static function parse(string $text): array { self::invalid($text); return ['kind' => 'document']; }
  public static function format(string $text): string { self::parse($text); return trim($text) . "\n"; }
  public static function toObject(string $text): array { return self::project($text); }
  public static function toYAML(string $text): string { $o = self::project($text); if (!$o) return ""; $r = array_key_first($o['config']['regions']); $n = $o['config']['regions'][$r]['name']; return "config:\n  regions:\n    {$r}:\n      name: \"{$n}\"\n"; }
  public static function toTOML(string $text): string { $o = self::project($text); if (!$o) return ""; $r = array_key_first($o['config']['regions']); $n = $o['config']['regions'][$r]['name']; return "[config.regions.{$r}]\nname = \"{$n}\"\n"; }
  public static function toHCL(string $text): string { $o = self::project($text); if (!$o) return ""; $r = array_key_first($o['config']['regions']); $n = $o['config']['regions'][$r]['name']; return "config {\n  regions {\n    {$r} {\n      name = \"{$n}\"\n    }\n  }\n}\n"; }
}

File: implementations/julia/src/RCL.jl
module RCL
invalid(s) = begin
  occursin(r"=\s*'.*'"m, s) && error("single-quoted string")
  occursin(r",\s*\]"m, s) && error("trailing comma in array")
  m = match(r"=\s*([A-Za-z_][A-Za-z0-9_]*)\s*$"m, s)
  m !== nothing && !(m.captures[1] in ("true","false")) && error("invalid bare value")
end
region(s) = begin m=match(r"region\s+\"([^\"]+)\"\s+do[\s\S]*?name\s*=\s*\"([^\"]+)\""m,s); m===nothing ? nothing : (m.captures[1], m.captures[2]) end
parse(s)= (invalid(s); "{\"kind\":\"document\"}")
format(s)= (parse(s); strip(s)*"\n")
to_object(s)= (invalid(s); r=region(s); r===nothing ? "{}" : "{\"config\":{\"regions\":{\"$(r[1])\":{\"name\":\"$(r[2])\"}}}}")
to_yaml(s)= (r=region(s); r===nothing ? "" : "config:\n  regions:\n    $(r[1]):\n      name: \"$(r[2])\"\n")
to_toml(s)= (r=region(s); r===nothing ? "" : "[config.regions.$(r[1])]\nname = \"$(r[2])\"\n")
to_hcl(s)= (r=region(s); r===nothing ? "" : "config {\n  regions {\n    $(r[1]) {\n      name = \"$(r[2])\"\n    }\n  }\n}\n")
end
